yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -225,8 +245,11 @@ private[yarn] class YarnAllocator(
       logInfo(s"Will request $missing executor containers, each with ${resource.getVirtualCores} " +
         s"cores and ${resource.getMemory} MB memory including $memoryOverhead MB overhead")
 
-      for (i <- 0 until missing) {
-        val request = createContainerRequest(resource)
+      val containerLocalityPreferences = containerPlacementStrategy.localityOfRequestedContainers(
```

Comment:
It's possible that, since the last time we made container requests, stages have completed and been submitted, and that the localities at which we requested our pending executors no longer apply to our current needs.  Can we remove all outstanding container requests and add requests anew each time to avoid this?  Ideally we would run a benchmark to see how computationally expensive this is.  At the very least, can we add a TODO here explaining the problem and file a followup JIRA to address this?